app/models/enterprise_relationship.rb

```diff
@@ -102,6 +104,16 @@ def revoke_all_child_variant_overrides
     child_variant_overrides.update_all(permission_revoked_at: Time.zone.now)
   end
 
+  def destroy_related_exchanges
+    ExchangeDeleter.new.delete(child, parent)
+  end
+
+  def check_order_cycle_permssions
+    return if has_permission?(:add_to_order_cycles)
+
+    destroy_related_exchanges
+  end
```

Comment:
I wonder if this would be better in EnterpriseRelationshipPermission.

Reply:
That could work. We need `destroy_related_exchanges` in `EnterpriseRelationship` to delete them if an `exchange` is destroyed, though. We would need to have similar code in two places, but it would be clearer.